### PR TITLE
Replace all OR statements with switch case

### PR DESCRIPTION
**Pull request by gavlyukovskiy**
Less crazy version of https://github.com/Breus/json-masker/pull/84

### CODE REVIEW

File: src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java [hunk 1]
```diff
@@ -58,12 +58,38 @@ public byte[] mask(byte[] input) {
...
-        if (AsciiCharacter.isSquareBracketOpen(maskingState.byteAtCurrentIndex())) {
-            visitArray(maskingState, keyMaskingConfig);
-        } else if (AsciiCharacter.isCurlyBracketOpen(maskingState.byteAtCurrentIndex())) {
-            visitObject(maskingState, keyMaskingConfig);
-        } else {
-            visitPrimitive(maskingState, keyMaskingConfig);
+        switch (maskingState.byteAtCurrentIndex()) {
+            case '[' -> visitArray(maskingState, keyMaskingConfig);
+            case '{' -> visitObject(maskingState, keyMaskingConfig);
+            case '-', '+', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '.' -> {
```
Comment (Breus) [inline src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java:64]: I don't think a number in valid JSON can start with a `.`  or a `+`.

Only `0-9` and `-` 

If your intent was to match on all numeric characters, you are missing `E` and `e`

File: src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java [hunk 2]
```diff
@@ -58,12 +58,38 @@ public byte[] mask(byte[] input) {
      */
     private void visitValue(MaskingState maskingState, @CheckForNull KeyMaskingConfig keyMaskingConfig) {
         skipWhitespaceCharacters(maskingState);
-        if (AsciiCharacter.isSquareBracketOpen(maskingState.byteAtCurrentIndex())) {
-            visitArray(maskingState, keyMaskingConfig);
-        } else if (AsciiCharacter.isCurlyBracketOpen(maskingState.byteAtCurrentIndex())) {
-            visitObject(maskingState, keyMaskingConfig);
-        } else {
-            visitPrimitive(maskingState, keyMaskingConfig);
+        switch (maskingState.byteAtCurrentIndex()) {
```
Comment (Breus) [inline src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java:61]: Might be nice to add a comment to each switch statement that it is deliberately a switch statement for improved performance

**Review by Breus — CHANGES_REQUESTED** [review]
Added 2 comments

File: src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java (hunk 2 above)
Comment (gavlyukovskiy) [reply src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java:62]: Done

File: src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java (hunk 1 above)
Comment (gavlyukovskiy) [reply src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java]: Done

**Review by Breus — APPROVED** [review]

**Pull request merged**